org.eclipse.xtext.xbase.ui.testing/src/org/eclipse/xtext/xbase/ui/testing/AbstractXbaseContentAssistTest.java

```diff
@@ -53,6 +55,82 @@
  */
 public abstract class AbstractXbaseContentAssistTest extends Assert implements ResourceLoadHelper, IJavaProjectProvider {
 
+	private static final boolean isJava11OrLater = determineJava11OrLater();
+
+	/**
+	 * @since 2.20
+	 */
+	public static boolean isJava11OrLater() {
+		return isJava11OrLater;
+	}
+
+	private static final boolean isJava12OrLater = determineJava12OrLater();
+
+	/**
+	 * @since 2.20
+	 */
+	public static boolean isJava12OrLater() {
+		return isJava12OrLater;
+	}
+
+	private static final boolean isJava13OrLater = determineJava13OrLater();
+
+	/**
+	 * @since 2.20
+	 */
+	public static boolean isJava13OrLater() {
+		return isJava13OrLater;
+	}
+
+	private static boolean determineJava11OrLater() {
+		String javaVersion = System.getProperty("java.version");
+		try {
+			Pattern p = Pattern.compile("(\\d+)(.)*");
+			Matcher matcher = p.matcher(javaVersion);
+			if (matcher.matches()) {
+				String first = matcher.group(1);
+				int version = Integer.parseInt(first);
+				return version >= 11;
+			}
+		} catch (NumberFormatException e) {
+			// ok
+		}
+		return false;
+	}
+
+	private static boolean determineJava12OrLater() {
+		String javaVersion = System.getProperty("java.version");
+		try {
+			Pattern p = Pattern.compile("(\\d+)(.)*");
+			Matcher matcher = p.matcher(javaVersion);
+			if (matcher.matches()) {
+				String first = matcher.group(1);
+				int version = Integer.parseInt(first);
+				return version >= 12;
+			}
+		} catch (NumberFormatException e) {
+			// ok
+		}
+		return false;
+	}
+
+	private static boolean determineJava13OrLater() {
+		String javaVersion = System.getProperty("java.version");
+		try {
+			Pattern p = Pattern.compile("(\\d+)(.)*");
+			Matcher matcher = p.matcher(javaVersion);
+			if (matcher.matches()) {
+				String first = matcher.group(1);
+				int version = Integer.parseInt(first);
+				return version >= 13;
```

Comment:
If I'm not wrong, these 3 methods are exactly the same but for the constant after `>=`... I guess it would be nice to have only one method taking the Java version as a parameter

Reply:
see https://github.com/eclipse/xtext/issues/1452